examples/chip-tool/commands/discover/DiscoverCommissionablesCommand.cpp

```diff
@@ -79,54 +84,54 @@ CHIP_ERROR DiscoverCommissionablesCommand::RunCommand()
 {
     mCommissioner = &CurrentCommissioner();
     mCommissioner->RegisterDeviceDiscoveryDelegate(this);
-    Dnssd::DiscoveryFilter filter(Dnssd::DiscoveryFilterType::kNone, (uint64_t) 0);
-    return mCommissioner->DiscoverCommissionableNodes(filter);
+    mFilter = { Dnssd::DiscoveryFilterType::kNone, (uint64_t) 0 };
+    return mCommissioner->DiscoverCommissionableNodes(mFilter);
 }
 
 CHIP_ERROR DiscoverCommissionableByShortDiscriminatorCommand::RunCommand()
 {
     mCommissioner = &CurrentCommissioner();
     mCommissioner->RegisterDeviceDiscoveryDelegate(this);
-    chip::Dnssd::DiscoveryFilter filter(chip::Dnssd::DiscoveryFilterType::kShortDiscriminator, mDiscriminator);
-    return mCommissioner->DiscoverCommissionableNodes(filter);
+    mFilter = { chip::Dnssd::DiscoveryFilterType::kShortDiscriminator, mDiscriminator };
+    return mCommissioner->DiscoverCommissionableNodes(mFilter);
 }
 
 CHIP_ERROR DiscoverCommissionableByLongDiscriminatorCommand::RunCommand()
 {
     mCommissioner = &CurrentCommissioner();
     mCommissioner->RegisterDeviceDiscoveryDelegate(this);
-    chip::Dnssd::DiscoveryFilter filter(chip::Dnssd::DiscoveryFilterType::kLongDiscriminator, mDiscriminator);
-    return mCommissioner->DiscoverCommissionableNodes(filter);
+    mFilter = { chip::Dnssd::DiscoveryFilterType::kLongDiscriminator, mDiscriminator };
+    return mCommissioner->DiscoverCommissionableNodes(mFilter);
 }
 
 CHIP_ERROR DiscoverCommissionableByCommissioningModeCommand::RunCommand()
 {
     mCommissioner = &CurrentCommissioner();
     mCommissioner->RegisterDeviceDiscoveryDelegate(this);
-    chip::Dnssd::DiscoveryFilter filter(chip::Dnssd::DiscoveryFilterType::kCommissioningMode);
-    return mCommissioner->DiscoverCommissionableNodes(filter);
+    mFilter = { chip::Dnssd::DiscoveryFilterType::kCommissioningMode, mCommissioningMode };
```

Comment:
This is a bit weird, because the `code` field is not really defined for a `kCommissioningMode`-type filter (though none of that is documented in `dnssd/Types.h`, sadly).

We should file an issue to document which fields make sense for which filter types, but for now: I suggest not putting `mCommissioningMode` in the filter at all.  The check up above can check `mCommissioningMode`.

Reply:
Agreed - this has been reverted in the latest commit.